### PR TITLE
machine-config-operator: Add e2e-aws-op job

### PR DESCRIPTION
**Pull request by cgwalters**
See:
https://github.com/openshift/machine-config-operator/pull/298
https://github.com/openshift/machine-config-operator/issues/257

### CODE REVIEW

**Comment by cgwalters** [comment]
/hold

Until the dependent PR merges.

File: ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml [hunk 1]
```diff
@@ -57,6 +57,120 @@ presubmits:
...
+  - agent: kubernetes
+    always_run: true
+    branches:
+    - master
+    context: ci/prow/e2e-aws-op
+    decorate: true
+    decoration_config:
+      skip_cloning: true
+    name: pull-ci-openshift-machine-config-operator-master-e2e-aws-op
+    rerun_command: /test e2e-aws-op
```
Comment (jlebon) [inline ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml]: Might want to also add the regex from #2576 here too.

File: ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml [hunk 2]
```diff
@@ -57,6 +57,120 @@ presubmits:
           name: prow-job-cluster-launch-installer-e2e
         name: job-definition
     trigger: ((?m)^/test( all| e2e-aws),?(\s+|$))
+  - agent: kubernetes
```
Comment (abhinavdahiya) [inline ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml]: why are we adding 2 jobs `e2e-aws-op` and `e2e-aws-operator` ?

Comment (cgwalters) [reply ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml]: Fixed

File: ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml (hunk 1 above)
Comment (cgwalters) [reply ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml]: AFAICS now that part isn't in the generated job (I think it's a configmap)?

**Comment by cgwalters** [comment]
Rebased :surfing_man: but depends on https://github.com/openshift/machine-config-operator/pull/309

File: ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml (hunk 1 above)
Comment (jlebon) [reply ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml]: It's higher up in that same file, no? It looks like it's working too. E.g. the CI status in https://github.com/openshift/machine-config-operator/pull/306 for that single commit says:

> ci/prow/e2e-aws — Skipped

Comment (cgwalters) [reply ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml]: Oh sorry I misunderstood; you want to also skip this test for docs (which makes sense).  I thought you were saying there'd be a merge conflict.

**Comment by cgwalters** [comment]
/hold cancel

Dependent PR merged, let's give this a try!

**Comment by cgwalters** [comment]
Looks like fallout from the Prow upgrade which was reverted.

/test config

**Pull request merged**